### ISSUE TITLE
Missed early opportunities for for-loop to list-comprehension conversion

### ISSUE THREAD

**Issue by gotdang** [sourcery-ai/sourcery#82]
### Issue description or question
There are three loops with the same structure in this function. 
Regardless of the order, only the last is identified as a list-comprehension conversion candidate.
If I convert the last one to a list comprehension, the earlier ones are still not identified as candidates.
If I delete the last one, only the new/remaining last one is then identified as a candidate.
```python
def address_filter(obj):
    if not obj:
        return ''
    # This loop is NOT identified as a list-comprehension candidate.
    line_1 = []
    for attr in ['street_address', 'street_address_1', 'street_address_2', 'address_1', 'address_2']:
        if getattr(obj, attr, None):
            line_1.append(getattr(obj, attr, ''))
    # This loop is NOT identified as a list-comprehension candidate.
    line_2 = []
    for attr in ['home', 'apartment', 'non-residential']:
        if getattr(obj, attr, None):
            line_2.append(getattr(obj, attr, ''))
    # This loop is identified as a list-comprehension candidate.
    line_3 = []  
    for attr in ['country']:  # I also tried with multiple attributes.
        if getattr(obj, attr, None):
            line_3.append(getattr(obj, attr, ''))

    return line_1 + line2 + line_3
```

### Sourcery Version
0.7.12

### Code editor or IDE name and version
VSCode 1.49.0

### OS name and version
Windows 10 running WSL/Debian 10

**Comment by Hellebore** [comment]
Thanks for raising - I'll take a look and see what's going on.

**Comment by Hellebore** [comment]
Our analysis is objecting to the reuse of `attr` after the loops.

This should be fine since it is being written with a new value rather than read, so we need to re-work it a bit to fix this.

**Comment by Hellebore** [comment]
This is now fixed in an upcoming release.

**Comment by Hellebore** [comment]
Addressed with release of 0.8.0
